src/oracles/ChainlinkBoundedCompositeOracle.sol

```diff
@@ -0,0 +1,275 @@
+    /// @param _lowerBound The lower bound for the primary oracle price
+    /// @param _upperBound The upper bound for the primary oracle price
+    /// @param _governor The address of the governor to own the oracle
+    constructor(
+        address _primaryOracle,
+        address _fallbackOracle,
+        int256 _lowerBound,
+        int256 _upperBound,
+        address _governor
+    ) {
+        primaryLBTCOracle = AggregatorV3Interface(_primaryOracle);
+        fallbackLBTCOracle = AggregatorV3Interface(_fallbackOracle);
+
+        require(
+            _lowerBound < _upperBound,
+            "ChainlinkBoundedCompositeOracle: Invalid bounds"
+        );
+
+        lowerBound = _lowerBound;
+        upperBound = _upperBound;
+
+        _transferOwnership(_governor);
+    }
+
+    /// @notice Get the latest price data from the oracle
+    /// @dev Checks primary oracle first, if within bounds combines with BTC price, otherwise uses fallback
+    /// @return roundId The round ID (always 0)
+    /// @return answer The price in USD with 18 decimals
+    /// @return startedAt The timestamp when the round started (always 0)
+    /// @return updatedAt The timestamp of the price update
+    /// @return answeredInRound The round ID in which the answer was computed (always 0)
+    function latestRoundData()
+        external
+        view
+        override
+        returns (
+            uint80 roundId,
+            int256 answer,
+            uint256 startedAt,
+            uint256 updatedAt,
+            uint80 answeredInRound
+        )
+    {
+        // Get primary price and scale it
+        (
+            ,
+            int256 primaryPrice,
+            uint256 primaryTimestamp,
+
+        ) = _getValidatedOracleData(primaryLBTCOracle);
+
+        uint8 primaryDecimals = primaryLBTCOracle.decimals();
+        primaryPrice = scalePrice(primaryPrice, primaryDecimals, decimals);
+
+        // Check primary lbtc/btc PoR exchange rate,
+        // if out of bounds, fall back to market rate oracle
+        if (primaryPrice < lowerBound || primaryPrice > upperBound) {
+            // fall back to market rate oracle if primary is out of bounds
+            (, primaryPrice, primaryTimestamp, ) = _getValidatedOracleData(
+                fallbackLBTCOracle
+            );
+            uint8 fallbackDecimals = fallbackLBTCOracle.decimals();
+            primaryPrice = scalePrice(primaryPrice, fallbackDecimals, decimals);
+        }
+
+        return (0, primaryPrice, 0, primaryTimestamp, 0);
+    }
+
+    /// @notice Get data about a specific round from the fallback oracle
+    /// @param _roundId The round ID to retrieve data for
+    /// @return roundId The round ID
+    /// @return answer The price
+    /// @return startedAt The timestamp when the round started
+    /// @return updatedAt The timestamp when the round was updated
+    /// @return answeredInRound The round ID in which the answer was computed
+    function getRoundData(
```

Comment:
my feeling is that it is better to remove this function. We don't need it for morpho, and if someone every does need it, the current implementation is misleading. If you do keep it, consider taking round data from primary and fallback, and apply the same logic as in get latest round.